app/DTO/TopAlbums.php

```diff
@@ -24,4 +25,31 @@ public function __construct(
 	) {
 		$this->shared_albums ??= new Collection();
 	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function toArray(): array
+	{
+		return [
+			'smart_albums' => $this->smart_albums->toArray(),
+			'tag_albums' => $this->tag_albums->map(fn ($a) => self::toAlbumDTOArray($a))->toArray(),
+			'albums' => $this->albums->map(fn ($a) => self::toAlbumDTOArray($a))->toArray(),
+			'shared_albums' => $this->shared_albums->map(fn ($a) => self::toAlbumDTOArray($a))->toArray(),
+		];
+	}
+
+	/**
+	 * Convert an Abstract album into it's DTO form.
+	 *
+	 * @param AbstractAlbum $abstractAlbum
+	 *
+	 * @return array resulting array
+	 */
+	private static function toAlbumDTOArray(AbstractAlbum $abstractAlbum): array
```

Comment:
As far as I can tell from code inspection
```suggestion
	private static function toAlbumDTOArray(BaseAlbum $baseAlbum): array
```
should suffice as the method is only called for regular albums, i.e. tag albums and albums, but never for smart albums. However, I haven't confirmed my suspicion with an IDE.

I don't know if this is somehow related to the reason why you moved the attribute `grants_full_photo_access` from `BaseAlbum` to `AbstractAlbum`. (See comment somewhere above.)

Reply:
This is now called an ALL albums (including smart)